src/BeaconConfig.jsx

```diff
@@ -28,4 +57,33 @@ export default class BeaconConfig extends Component {
       </div>
     );
   }
+
+  storeHash(hash) {
+    const transaction = this.state.database.transaction(['beacons'], 'readwrite');
+    transaction.objectStore('beacons').add(true, hash);
+
+    this.setState({ dbCache: { ...this.state.dbCache, [hash]: hash }});
+  }
+
+  loadHash(hash, callback) {
+    if (hash in this.state.dbCache && callback) {
+      callback(this.state.dbCache[hash] !== null);
+    } else if (!this.state.database) {
+      // Database is not open yet so queue request
+      this.state.requestQueue.push({ hash: hash, callback: callback });
+    } else {
+      const transaction = this.state.database.transaction(['beacons']);
+      const objectStore = transaction.objectStore('beacons');
+      objectStore.get(hash).onsuccess = (event) => {
+        this.state.dbCache[hash] = event.target.result ? hash : null;
+        if (callback) {
+          callback(event.target.result);
+        }
+      };
+    }
+  }
+
+  checkHash(hash) {
+    return (hash in this.state.dbCache) && this.state.dbCache[hash];
```

Comment:
Returns `false` if we haven't cached the value, `null` if it wasn't found and the hash otherwise. These are then translated to the "enum" in `Beacon.jsx` although I kind of wonder if we shouldn't store those `HASH_CHECK_*` constants somewhere centrally and use them in both files.